vm/error_test.go

```diff
@@ -112,3 +98,56 @@ func TestUnsupportedMethodError(t *testing.T) {
 		}
 	}
 }
+
+func TestArgumentError(t *testing.T) {
```

Comment:
@Maxwell-Alexius  @hachi8833 Can you help me check if these test cases are enough?